### PR TITLE
rebased Brainfuck example with AATree removed for now and CI passing.

### PR DESCRIPTION
**Pull request by ceberly**

### CODE REVIEW

File: Brainfuck/spark/program.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+source: src/spark.adb
+
+prepare:
+  - alr build 
+  - alr gnatprove
+
+run: ./bin/spark {PROGRAM}
+
+authors:
+  - eatonphil
```
Comment (eatonphil) [inline Brainfuck/spark/program.yaml]: No need to have me in here, it's understood that the different languages are ports. Just your name is fine! :)

File: Brainfuck/spark/spark.gpr [hunk 2]
```diff
@@ -0,0 +1,22 @@
+with "config/spark_config.gpr";
```
Comment (eatonphil) [inline Brainfuck/spark/spark.gpr]: Is there a way to get it to build without needing this script? I.e. using only command line arguments to `alr` I'm guessing?

File: Brainfuck/spark/alire.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name = "spark"
```
Comment (eatonphil) [inline Brainfuck/spark/alire.toml]: None of the other ports make use of package managers. Can we do this without package managers?

Comment (eatonphil) [reply Brainfuck/spark/alire.toml]: If that means dropping `gnatprove` then maybe we can just leave that as a note in program.yaml.

I've struggled that while it's useful to also see a non-trivial project (such as using more than one file and bringing in a dependency), I don't really want that complexity for now across the board.

File: Brainfuck/spark/spark.gpr (hunk 2 above)
Comment (ceberly) [reply Brainfuck/spark/spark.gpr]: Definitely. `alr` is the `cargo` of the ada world. I can build manually, *or* if you are ok having the files generated just at build time (similar to `go.mod` etc), then I can create the alr project in-place then we can keep leveraging the alire github action.

File: Brainfuck/spark/alire.toml (hunk 3 above)
Comment (ceberly) [reply Brainfuck/spark/alire.toml]: see above comment.

File: Brainfuck/spark/program.yaml (hunk 1 above)
Comment (ceberly) [reply Brainfuck/spark/program.yaml]: 🙏

File: Brainfuck/spark/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+TLDR; usage
+
+- Follow the instructions for downloading and setting up Alire https://alire.ada.dev
+- `alr gnatprove` will run the proofs.
+- `alr run --args=../_tests/helloworld.bf` will run the example program.
```
Comment (eatonphil) [inline Brainfuck/spark/README.md]: Last thing, could you move these comments into programy.yaml in the `notes` objects? https://github.com/eatonphil/tinyprograms/blob/main/Brainfuck/freepascal/program.yaml#L7

And we can get rid of the README.

Comment (eatonphil) [reply Brainfuck/spark/README.md]: Those notes get rendered into the UI: https://tinyprograms.org/Brainfuck/freepascal.html

**Comment by eatonphil** [comment]
Huzzah!

**Comment by eatonphil** [comment]
It's live!

https://tinyprograms.org/Brainfuck/spark.html

**Pull request merged**